mautic/mautic
# Fixed test connection button for SMTP that the fix for SparkPost brok…

| Q | A |
| --- | --- |
| Bug fix? | Y |
| New feature? | m |
| Related user documentation PR URL | n |
| Related developer documentation PR URL | n |
| Issues addressed (#s or URLs) |  |
| BC breaks? | n |
| Deprecations? | n |
#### Description:
#2708 fixed an issue with SparkPost but introduced an issue for SMTP because of Swift_Transport_EsmtpTransport uses a magic method making is_callable true. The fix is to simply wrap in a try/catch in case magic methods throw exceptions.
#### Steps to test this PR:
1. Go to Configuration and choose "Other SMTP Server" for the email transport
2. Click Test Connection - an exception should not bubble up (should get success if using valid smtp credentials or SMTP error.
3. Choose SparkPost and check connection - enter any key into the API key and click Test Connection. Should get Forbidden (which notes that the API key was set; doing the same with an empty API key should result in a empty API key error.
#### Steps to reproduce the bug:
1. Same as above but SMTP will throw an uncaught exception
